provider/pkg/provider/provider.go

```diff
@@ -2801,8 +2801,31 @@ func mapReplStripSecrets(v resource.PropertyValue) (interface{}, bool) {
 	return nil, false
 }
 
+// mapReplUnderscoreToDash is needed to work around cases where SDKs don't allow dashes in variable names, and so the
+// parameter is renamed with an underscore during schema generation. This function normalizes those keys to the format
+// expected by the cluster.
+func mapReplUnderscoreToDash(v string) (string, bool) {
```

Comment:
Are these comprehensive? Is there a reference to where these fields are defined we can add to the code here?

Reply:
Yeah, I verified that these were the only instances in the schema with this problem. The conversion happens in the [schema generation here](https://github.com/pulumi/pulumi-kubernetes/blob/fa399ca3a3eae8401de717c67dc351e6932d6d91/provider/pkg/gen/typegen.go#L564-L565)

I checked the conversions against the comments in the OpenAPI spec.

Reply:
I made the conversion explicit on the schema generation side as well to verify that we catch any similar problems that are added in the future.